Bugfix: el jugador deja hueco al agarrarse a un objeto perpendicularmente

Adrian-de-Lucas-Gomez opened the pull request:
Se ha arreglado el problema de dejar hueco al pegarnos a una pared cuando íbamos perpendicular a ella o en diagonal. Ahora te quedas bien pegadito.

Adrian-de-Lucas-Gomez committed:
Desarmar jugadores

Se ha creado una constante "DISARM_IMPACT" la cual marca el limite para poder ser golpeado por un arma y que no te pase nada. Si la velocidad que lleva lo sobrepasa te quedarás desarmado en el caso de que tuvieras un arma.
Por ahora solo te desarman las Weapon puras como la pelota saltarina por el problema de Weapon y sus clases hijas.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Desarmar por colision de objetos lanzables

Para evitar problemas con los diferentes tipos de armas se ha optado por diferenciar los obketos si tienen el componente ThrownByPlayer ya que esto nos permite saber de forma implicita que se lanza y como sabemos los objetos que lanzamos (Los cuales todos son armas) y por lo tanto los que te pueden desarmar.

Se ha puesto un margen para la velocidad del impacto para evitar desarmarte al tocar con un arma que estaba estática.

Adrian-de-Lucas-Gomez committed:
Ajustes en sprites para coherencia

Simplemente para adaptarlos a los acpectos que tienen las armas cuando están flotando

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Cambios al router

Solo un sprite mas claro para que se vea mejor

Adrian-de-Lucas-Gomez committed:
Eliminado en hueco que se dejaba al agarrase

Ya se queda bien pegadito a la pared. Lo he probado en la caldera que es la situación mas compleja.

Adrian-de-Lucas-Gomez committed:
Mejoras al platano

Si han refinado los sprites y se han hecho todos sobre el mismo diseño de Hands.

Tambien he quitado una cosa innecesaria que puse en el agarre

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

EstebenR reviewed (changes requested):
Te puedes salir del mapa abusando el agarre, paso video por discord

Adrian-de-Lucas-Gomez committed:
Prueba para evitar clipping

Es una solucion y ayuda a la hora de hacer impulsos desde la pared pues de aleja un poco de ella.

Adrian-de-Lucas-Gomez committed:
Creo que ahora si (Los agarres)

Esto es un sufrimiento pero creo que ya estando en una sola pared no se puede clipear al otro lado. No he cambiado el grosor de las paredes pero como sea necesario se hará.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Por Jesus bendito ya va bien

He dejado lo de que al impulsarte te separe un poco de la pared, ya no es necesario pero creo que beneficia a la movilidad evitando que roces con la pared.

Adrian-de-Lucas-Gomez committed:
Eliminada comprobacion con normales

Era demasiado restrictivo ya que hay paredes en diagonal

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

EstebenR reviewed (changes requested):
La distancia de 'reajuste' sigue siendo demasiada
Hay veces que si tu collider está girado (no perpendicular ni paralelo) a la pared sigue habiendo hueco
Por el reajuste si te agarras justo en una de las esquinas del mapa te mete en la pared contigua

Adrian-de-Lucas-Gomez committed:
Merge de las manitas

Adrian-de-Lucas-Gomez committed:
Ahora si que si

Creo que ya por fin he dado con la forma idonea para hacer los acercamientos a la pared. Basicamente usando las normales del plano pero en sentido contrario.

Adrian-de-Lucas-Gomez committed:
Mejoras del agarre

Se ha establecido un parametro de tolerancia de ángulos para evitar acercar demasiado al jugador si tiene un angulo cercano al paralelo a la pared

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Variables con nombres buenos

Que sino Cleon nos va a pegar

EstebenR reviewed (changes requested):
Lo que dice Jorge de no poder impulsarse hacia donde estas agarrado es buena idea, soluciona varios de los bugs que suceden por lo mismo

hay varios numeros magicos que no sé que son xd
